src/evaluators/ForStatement.js

```diff
@@ -216,7 +216,7 @@ function ForBodyEvaluation(
       invariant(result instanceof JoinedAbruptCompletions);
       invariant(result.consequent instanceof ContinueCompletion || result.alternate instanceof ContinueCompletion);
       joinedEffects = extractAndSavePossiblyNormalCompletion(ContinueCompletion, result);
-      result = joinedEffects.data[0];
+      result = joinedEffects.result;
```

Comment:
Here is an example where we are already inconsistent and it doesn't seem quite so bad.